core/src/replay_stage.rs

```diff
@@ -4035,6 +3988,95 @@ impl ReplayStage {
         duplicate_confirmed_forks
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn handle_new_root_all(
```

Comment:
I like this change, but can we use better names than `handle_new_root_all` so it's obvious what's the difference between it and `handle_new_root`?

Reply:
For example, `check_and_handle_new_root` is also a bad name, but at least it describes that we are confirming we are having a new root now. We should really refactor this code someday.